presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -85,30 +85,45 @@ protected PlanNode visitPlan(
                 RewriteContext<Multimap<WindowNode.Specification, WindowNode>> context)
```

Comment:
Ok, I agree with you. Lets not complicate this now. I would just create an issue about it, in case such use cases occur.

Although, it is hard to me to believe that is NP-hard. Anyway, nevermind lets put it aside.